src/containers/Auth/LoginScreen.js

```diff
@@ -27,21 +28,30 @@ export const LoginScreen = () => {
   }, [error])
 
   const handleLogin = (values) => {
-    API.postLogin(values).then((response) => {
-      if (response.statusCode!==200) {
-        if (response.data.non_field_errors) {
-          setError(response.data.non_field_errors[0])
-        } else if (response.data.username) {
-          setError('Username or E-mail address: ' + response.data.username[0])
-        } else if (response.data.password) {
-          setError('Password: ' + response.data.password[0])
-        } else {
-          setError('Something went wrong, please try again.')
+    setSubmitPending(true)
+    return API.postLogin(values)
+      .then((response) => {
+        const {statusCode, data} = response
+        if (statusCode === 200) {
+          signIn(data.token)
+        } else if (statusCode === 400) {
+          let errorMessage
+          if (data.non_field_errors) {
+            setError(data.non_field_errors[0])
```

Comment:
does this work? I think if you handle errors with the catch block below, we should not setError and trigger a rerender but rather `errorMessage = data.non_field_errors[0]`.

maybe it is better to throw the error directly with `throw new Error('some error message string'). With this we immediately enter the `catch` block. Does that make sense?

Reply:
did test but i'll check it again, I just wanted it to be same format as our other submits but didn't think too much about it